.github/actions/setup-linux/action.yml

```diff
@@ -104,3 +105,28 @@ runs:
 
         echo "${RESOLVED_IP} ${PT_DOMAIN}" | sudo tee -a /etc/hosts
         cat /etc/hosts
+
+    - name: Check that the docker daemon is running
```

Comment:
I feel like while this is good here it might be better suited to be on the runner itself in a pre-job script.

Docs for reference: https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/running-scripts-before-or-after-a-job

Reply:
Good point. I would like to keep it here for now because it makes it easier to debug the logs. Tbh, I haven't seen any example of the docker daemon access issue for last couple of days. Even the HUD is green ([all jobs passed on the first try](https://github.com/pytorch/pytorch/actions/runs/8828092744/job/24236434390)) and perhaps we don't need this check at all. If the problem continues to occur and we need this logic, I will move the script to the runner.

Reply:
With the ARC jobs moved to `unstable`, they no longer run on normal PRs by default. So right now our ARC usage is way lower than before which may be contributing to the lack of docker connection failures.

Until we know for sure what caused the connection to fail, I would assume that the underlying cause still exists and we should plan accordingly